strings.go

```diff
@@ -0,0 +1,328 @@
+			&ast.Ident{Name: "err"},
+		},
+		Tok: token.DEFINE,
+		Rhs: []ast.Expr{
+			&ast.CallExpr{
+				Fun: &ast.SelectorExpr{
+					X:   &ast.Ident{Name: "cipher"},
+					Sel: &ast.Ident{Name: "NewGCM"},
+				},
+				Args: []ast.Expr{
+					&ast.Ident{Name: "block"},
+				},
+			},
+		},
+	}
+
+	plaintextStmt = &ast.AssignStmt{
+		Lhs: []ast.Expr{
+			&ast.Ident{Name: "plaintext"},
+			&ast.Ident{Name: "err"},
+		},
+		Tok: token.DEFINE,
+		Rhs: []ast.Expr{
+			&ast.CallExpr{
+				Fun: &ast.SelectorExpr{
+					X:   &ast.Ident{Name: "aesgcm"},
+					Sel: &ast.Ident{Name: "Open"},
+				},
+				Args: []ast.Expr{
+					&ast.Ident{Name: "nil"},
+					&ast.SliceExpr{
+						X: &ast.Ident{Name: "ciphertext"},
+						High: &ast.BasicLit{
+							Kind:  token.INT,
+							Value: "12",
+						},
+					},
+					&ast.SliceExpr{
+						X: &ast.Ident{Name: "ciphertext"},
+						Low: &ast.BasicLit{
+							Kind:  token.INT,
+							Value: "12",
+						},
+					},
+					&ast.Ident{Name: "nil"},
+				},
+			},
+		},
+	}
+
+	returnStmt = &ast.ReturnStmt{
+		Results: []ast.Expr{
+			&ast.CallExpr{
+				Fun: &ast.Ident{Name: "string"},
+				Args: []ast.Expr{
+					&ast.Ident{Name: "plaintext"},
+				},
+			},
+		},
+	}
+)
+
+func decErrStmt() *ast.IfStmt {
+	return &ast.IfStmt{
+		Cond: &ast.BinaryExpr{
+			X:  &ast.Ident{Name: "err"},
+			Op: token.NEQ,
+			Y:  &ast.Ident{Name: "nil"},
+		},
+		Body: &ast.BlockStmt{
+			List: []ast.Stmt{
+				&ast.ExprStmt{
+					X: &ast.CallExpr{
+						Fun: &ast.Ident{Name: "panic"},
+						Args: []ast.Expr{
+							&ast.BinaryExpr{
+								X: &ast.BasicLit{
+									Kind:  token.STRING,
+									Value: "\"[garble] Literal couldn't be decrypted: \"",
+								},
+								Op: token.ADD,
+								Y: &ast.CallExpr{
+									Fun: &ast.SelectorExpr{
+										X:   &ast.Ident{Name: "err"},
+										Sel: &ast.Ident{Name: "Error"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func funcStmt() *ast.FuncDecl {
+	stmts := []ast.Stmt{
+		aesCipherStmt,
+		decErrStmt(),
+		aesGcmCipherStmt,
+		decErrStmt(),
+		plaintextStmt,
+		decErrStmt(),
+		returnStmt,
+	}
+
+	return &ast.FuncDecl{
+		Name: &ast.Ident{Name: "garbleDecrypt"},
+		Type: &ast.FuncType{
+			Params: &ast.FieldList{
+				List: []*ast.Field{
+					{
+						Names: []*ast.Ident{{Name: "ciphertext"}},
+						Type: &ast.ArrayType{
+							Elt: &ast.Ident{Name: "byte"},
+						},
+					},
+				},
+			},
+			Results: &ast.FieldList{
+				List: []*ast.Field{
+					{
+						Type: &ast.Ident{Name: "string"},
+					},
+				},
+			},
+		},
+		Body: &ast.BlockStmt{
+			List: stmts,
+		},
+	}
+
+}
+
+func ciphertextStmt(ciphertext []byte) *ast.CallExpr {
+	ciphertextLit := byteToByteLit(ciphertext)
+
+	return &ast.CallExpr{
+		Fun: &ast.Ident{Name: "garbleDecrypt"},
+		Args: []ast.Expr{
+			ciphertextLit,
+		},
+	}
+}
+
+func byteToByteLit(buffer []byte) *ast.CallExpr {
+
+	hexstr := hex.EncodeToString(buffer)
+
+	var escHexStr = "\""
```

Comment:
adding to a string repeatedly in Go is a very bad idea - you'll allocate a new string and copy at each iteration. use a `strings.Builder` instead.